src/frontend/apps/lti_site/apps/classroom/components/UploadDocuments/index.tsx

```diff
@@ -89,46 +91,61 @@ const UploadDocumentsRow = ({
   }
 
   return (
-    <Box fill background="#F9FBFD" pad="small" round="small" direction="row">
-      <Box justify="start" flex>
-        <Text weight="bold">{truncateFilename(filename, 40)}</Text>
-      </Box>
-      {uploadingObjectId && (
-        <React.Fragment>
-          <UploadableObjectProgress objectId={uploadingObjectId} />
-          {intl.formatMessage(messages.uploadingFile)}
-        </React.Fragment>
-      )}
-      {uploadState && (
-        <Box justify="end" flex="shrink" direction="row">
-          {uploadState === 'ready' ? (
-            <Box direction="row" align="center" gap="small">
-              {isDefault ? (
-                <ValidSVG iconColor="brand" height="20px" width="20px" />
-              ) : (
-                <Button
-                  alignSelf="start"
-                  onClick={setDefaultDocument}
-                  title={intl.formatMessage(messages.setDefaultDocument)}
-                >
-                  <ValidSVG iconColor="light-5" height="20px" width="20px" />
-                </Button>
-              )}
-              <Anchor
-                download
-                a11yTitle={intl.formatMessage(messages.downloadButtonLabel)}
-                label={intl.formatMessage(messages.downloadButtonLabel)}
-                style={{ fontFamily: 'Roboto-Medium' }}
-                title={intl.formatMessage(messages.downloadButtonLabel)}
-                href={url}
-              />
-            </Box>
-          ) : (
-            <Text>{uploadState}</Text>
-          )}
+    <Grid fill>
+      <Box
+        fill
+        background="#F9FBFD"
+        pad="small"
+        round="small"
+        direction={uploadingObjectId ? 'column' : 'row'}
+      >
+        <Box
+          justify={uploadingObjectId ? 'center' : 'start'}
+          flex={!uploadingObjectId}
+        >
+          <Text as="div" weight="bold">
+            {truncateFilename(filename, 40)}
```

Comment:
Why not using the truncate props of grommet ?

Reply:
Because it's a custom one, which keeps the file extension displayed.